src/System.Security.Claims/src/System/Security/Claims/RoleClaimProvider.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+// 
+//
+// RoleClaimProvider.cs
+//
+
+using System.Collections.Generic;
+
+namespace System.Security.Claims
+{
+    /// <summary>
+    /// This internal class is used to wrap role claims that can be set on GenericPrincipal.  They need to be kept distinct from other claims.
+    /// ClaimsIdentity has a property the holds this type.  Since it is internal, few checks are
+    /// made on parameters.
+    /// </summary>    
+
+    [System.Runtime.InteropServices.ComVisible(false)]
+    internal class RoleClaimProvider
+    {
+        string m_issuer;
+        string[] m_roles;
+        ClaimsIdentity m_subject;
```

Comment:
This class also doesn't currently follow the coding guidelines as these fields should be explicitly marked private and should have a "_" prefix instead of "m_".  See https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md